app-editors/helix/helix-23.03.ebuild

```diff
@@ -0,0 +1,337 @@
+	wasm-bindgen-backend-0.2.84
+	wasm-bindgen-macro-0.2.84
+	wasm-bindgen-macro-support-0.2.84
+	wasm-bindgen-shared-0.2.84
+	which-4.4.0
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+	windows-0.43.0
+	windows-0.47.0
+	windows-sys-0.45.0
+	windows-targets-0.42.2
+	windows-targets-0.47.0
+	windows_aarch64_gnullvm-0.42.2
+	windows_aarch64_gnullvm-0.47.0
+	windows_aarch64_msvc-0.42.2
+	windows_aarch64_msvc-0.47.0
+	windows_i686_gnu-0.42.2
+	windows_i686_gnu-0.47.0
+	windows_i686_msvc-0.42.2
+	windows_i686_msvc-0.47.0
+	windows_x86_64_gnu-0.42.2
+	windows_x86_64_gnu-0.47.0
+	windows_x86_64_gnullvm-0.42.2
+	windows_x86_64_gnullvm-0.47.0
+	windows_x86_64_msvc-0.42.2
+	windows_x86_64_msvc-0.47.0
+	winnow-0.4.1
+"
+
+# Download the same tree-sitter commit that is used in the Helix release
+declare -A GIT_CRATES=(
+	[tree-sitter]="https://github.com/tree-sitter/tree-sitter;c51896d32dcc11a38e41f36e3deb1a6a9c4f4b14;tree-sitter-%commit%/lib"
+)
+
+inherit bash-completion-r1 cargo desktop xdg-utils
+
+DESCRIPTION="A post-modern text editor."
+HOMEPAGE="
+	https://helix-editor.com/
+	https://github.com/helix-editor/helix
+"
+SRC_URI="
+	https://github.com/helix-editor/helix/releases/download/${PV}/${P}-source.tar.xz -> ${P}.tar.xz
+	$(cargo_crate_uris)
+"
+
+LICENSE="0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD Boost-1.0 ISC MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc +grammar"
+
+QA_FLAGS_IGNORED="
+	usr/bin/hx
+	usr/share/helix/runtime/grammars/.*\.so
+"
+
+DOCS=(
+	README.md
+	CHANGELOG.md
+	book/
+	docs/
+)
+
+S="${WORKDIR}"
+
+src_prepare() {
+	# A small hack needed to override the patch in the main Cargo.toml
+	sed -i 's|https://github.com/tree-sitter/tree-sitter|crates-io|' "${WORKDIR}/cargo_home/config"
+	eapply_user
+}
+
+src_compile() {
+	use !grammar && local -x HELIX_DISABLE_AUTO_GRAMMAR_BUILD=1
+
+	cargo_src_compile
+}
+
+src_install() {
+	cargo_src_install --path helix-term
+
+	rm -rf runtime/grammars/.gitkeep || die
+	rm -rf runtime/grammars/sources || die
+
+	insinto /usr/share/helix
+	doins -r runtime
+
+	use doc && dodoc -r "${DOCS[@]}"
+
+	doicon -s 256x256 contrib/${PN}.png
+	domenu contrib/Helix.desktop
+
+	insinto /usr/share/metainfo
+	doins contrib/Helix.appdata.xml
+
+	newbashcomp contrib/completion/hx.bash hx
+
+	insinto /usr/share/zsh/site-functions
+	newins contrib/completion/hx.zsh _hx
+
+	insinto /usr/share/fish/vendor_completions.d
+	doins contrib/completion/hx.fish
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+	einfo "The runtime files (grammars, queries, themes) are"
+	einfo "installed in '/usr/share/helix/runtime'. You may want to set"
+	einfo "HELIX_RUNTIME=/usr/share/helix/runtime or create symlinks"
+	einfo "in your user runtime folder that point to '/usr/share/helix/runtime'."
+	einfo "See https://docs.helix-editor.com/install.html?highlight=runtime#linux-and-macos"
+}
```

Comment:
Maybe it would be worth dropping a file in `/etc/env.d` to set `HELIX_RUNTIME` globally?

Reply:
A very interesting idea! Never would have thought of that. I never did something like this, so please take a look at  42b072b. I really appreciate your feedback by the way :)